test/boot_arch_test.rb

```diff
@@ -0,0 +1,170 @@
+require_relative "test_helper"
+
+Yast.import "BootArch"
+
+describe Yast::BootArch do
+  subject { Yast::BootArch }
+
+  def stub_arch(arch)
+    Yast.import "Arch"
+
+    allow(Yast::Arch).to receive(:architecture).and_return(arch)
+  end
```

Comment:
I know you are not a big fan of the excessive nesting usually introduced by ```context```, but I'd say that this one of those cases in which the RSpec good practices would suggest using ```let```
```ruby
describe Yast::BootArch do
  subject { Yast::BootArch }
  before do
    allow(Yast::Arch).to receive(:architecture).and_return(arch)
  end

  describe ".method" do
    context "on x86_64" do
      let(:arch) { "x86_64" }
    end
    ...
  end
end
```

But I can live with the current approach as well.